### PR TITLE
Dev 244 add a new check

### PR DESCRIPTION
**Pull request by iranimij**

### CODE REVIEW

File: etc/di.xml [hunk 1]
```diff
@@ -3,4 +3,6 @@
     <type name="Magento\Store\Model\Store">
         <plugin name="hide_default_store_code" type="CustomGento\DefaultStoreCodeRemover\Plugin\Store\HideDefaultStoreCodePlugin"/>
     </type>
+
```
Comment (norgeindian) [inline etc/di.xml:6]: Please remove the empty line

File: Helper/StoreUrl.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+declare(strict_types=1);
+
+namespace CustomGento\DefaultStoreCodeRemover\Helper;
+
+use Magento\Store\Api\Data\StoreInterface;
+use Magento\Store\Model\Store;
+use MageWorx\SeoBase\Helper\StoreUrl as MageworxStoreUrl;
+
+if (!class_exists(MageworxStoreUrl::class)) {
```
Comment (norgeindian) [inline Helper/StoreUrl.php]: As Simon pointed out, this is not necessary, as we will never come into this class, if the original one does not exist, right?
So let's remove it, I would say.

**Review by norgeindian — CHANGES_REQUESTED** [review]

**Review by norgeindian — APPROVED** [review]

**Comment by pknap** [comment]
Hi @iranimij @norgeindian,

this change references 3rd party module

`<preference for="MageWorx\SeoBase\Helper\StoreUrl" type="CustomGento\DefaultStoreCodeRemover\Helper\StoreUrl"/>`

which if not installed causes exceptions during di:compile:

`[localhost] There is an error in /builds/xxx/vendor/customgento/module-default-store-code-remover-m2/Helper/StoreUrl.php at line: 11
[localhost] Class "MageWorx\SeoBase\Helper\StoreUrl" not found#0 /builds/xxx/vendor/composer/ClassLoader.php(582): include()` 

Should this really be a part of this module?

**Comment by sprankhub** [comment]
Ohhh that's bad, @pknap, sorry for the inconvenience. Actually, I didn't think this leads to issues. I thought Magento is smart enough to ignore the preference when the overridden class does not exist. I would love to fix the issue inside the extension itself if possible, so that it just works even if the MageWorx extension is used. I can currently only think of conditionally overriding the class like this:

    if (class_exists(MageworxStoreUrl::class)) {
        class StoreUrl extends MageworxStoreUrl
        {
            [...]
        }
    } else {
        class StoreUrl {
    
        }
    }

It's not nice, but fixes the issue.

By chance, do you have any other idea?

**Comment by pknap** [comment]
Thank you for replying @sprankhub. 

I think the best practice would be not to include this preference in this module, but rather create a separate module which depends on `MageWorx\SeoBase` and `CustomGento_DefaultStoreCodeRemover` but I understand that would be overkill from your perspective. 

I think in this case there is no other way than what you have proposed. I had an idea that maybe virtual "placeholder" class could be created for `MageworxStoreUrl` but I don't think it would be possible to do it conditionally in di.xml, so class_exists seems to be only option under this circumstances, unless I am not aware of something.

**Comment by norgeindian** [comment]
@sprankhub , I tried your approach, but it does not work:
```
if (class_exists(MageWorx\SeoBase\Helper\StoreUrl::class)) {
    class StoreUrl extends MageworxStoreUrl
    {
```
directly leads to the di compilation issue, so I don't see any way of implementing that properly and would go for the idea of @pknap and create an additional module.
What do you think?

**Comment by sprankhub** [comment]
@norgeindian, I don't get an issue with `bin/magento setup:di:compile` using this code:

    <?php

    declare(strict_types=1);

    namespace CustomGento\DefaultStoreCodeRemover\Helper;

    use Magento\Store\Api\Data\StoreInterface;
    use Magento\Store\Model\Store;
    use MageWorx\SeoBase\Helper\StoreUrl as MageworxStoreUrl;

    if (class_exists(MageworxStoreUrl::class)) {
        class StoreUrl extends MageworxStoreUrl
        {
            protected function isUseStoreCodeInUrl(StoreInterface $store): bool
            {
                if ($store->getCode() !== Store::ADMIN_CODE && $store->isDefault()) {
                    return false;
                }

                $storeId = (int)$store->getId();

                return !($store->hasDisableStoreInUrl() && $store->getDisableStoreInUrl())
                    // @phpstan-ignore-next-line
                    && $this->configDataLoader->getConfigValue(Store::XML_PATH_STORE_IN_URL, $storeId);
            }
        }
    } else {
        class StoreUrl
        {

        }
    }

Am I missing something?

**Comment by norgeindian** [comment]
@sprankhub , how do you reference this file in your `di.xml`?

**Comment by sprankhub** [comment]
@norgeindian, exactly as before:

    <preference for="MageWorx\SeoBase\Helper\StoreUrl" type="CustomGento\DefaultStoreCodeRemover\Helper\StoreUrl"/>

**Pull request merged**